wire-ios-sync-engine/Tests/Source/Synchronization/SyncAgentTests.swift

```diff
@@ -16,37 +16,48 @@
 // along with this program. If not, see http://www.gnu.org/licenses/.
 //
 
+import WireDomain
 import XCTest
 @testable import WireDataModelSupport
 @testable import WireDomainSupport
 @testable import WireSyncEngine
 
-final class SyncAgentTests: XCTestCase {
+final class SyncAgentTests: XCTestCase, InitialSyncProvider, IncrementalSyncProvider {
 
     var sut: SyncAgent!
     var lastUpdateEventIDRepository: MockLastEventIDRepositoryInterface!
-    var initialSyncBuilder: MockInitialSyncBuilderProtocol!
     var legacySyncStatus: MockSyncStatusProtocol!
     var initialSync: MockInitialSyncProtocol!
+    var incrementalSync: MockIncrementalSyncProtocol!
 
     override func setUp() {
         lastUpdateEventIDRepository = MockLastEventIDRepositoryInterface()
-        initialSyncBuilder = MockInitialSyncBuilderProtocol()
         legacySyncStatus = MockSyncStatusProtocol()
         initialSync = MockInitialSyncProtocol()
+        incrementalSync = MockIncrementalSyncProtocol()
         sut = SyncAgent(
             lastUpdateEventIDRepository: lastUpdateEventIDRepository,
-            initialSyncBuilder: initialSyncBuilder,
+            initialSyncProvider: self,
+            incrementalSyncProvider: self,
             legacySyncStatus: legacySyncStatus
         )
-        initialSyncBuilder.buildInitialSync_MockValue = initialSync
     }
 
     override func tearDown() {
         sut = nil
         lastUpdateEventIDRepository = nil
-        initialSyncBuilder = nil
         legacySyncStatus = nil
+        initialSync = nil
+        incrementalSync = nil
+        DeveloperFlag.newInitialSync.enable(false, storage: .temporary())
```

Comment:
though: this is not necessary, if you set to temporary

```suggestion
        DeveloperFlag.newInitialSync.enable(false, storage: .standard())
```

Reply:
hmm... I saw some failed tests that were only affected when running the whole sync engine suite, I assumed it was because of some sticky flags.